mlflow/utils/virtualenv.py

```diff
@@ -420,14 +421,28 @@ def _get_or_create_virtualenv(  # noqa: D417
             pyenv_root_path.mkdir(parents=True, exist_ok=True)
             pyenv_root_dir = str(pyenv_root_path)
 
-    virtual_envs_root_path = (
-        Path(env_root_dir) / _VIRTUALENV_ENVS_DIR
-        if env_root_dir is not None
-        else Path(_get_mlflow_virtualenv_root())
-    )
     virtual_envs_root_path.mkdir(parents=True, exist_ok=True)
     env_name = _get_virtualenv_name(python_env, local_model_path, env_id)
     env_dir = virtual_envs_root_path / env_name
+    try:
+        env_dir.exists()
+    except PermissionError:
+        if is_in_databricks_runtime():
+            # Updating env_name only doesn't work because the cluster may not have
+            # permission to access the original virtual_envs_root_path
+            virtual_envs_root_path = (
+                Path(env_root_dir) / f"{_VIRTUALENV_ENVS_DIR}_{uuid.uuid4().hex[:4]}"
+            )
+            virtual_envs_root_path.mkdir(parents=True, exist_ok=True)
+            env_dir = virtual_envs_root_path / env_name
+        else:
+            _logger.warning(
+                f"Existing virtual environment directory {env_dir} cannot be accessed "
+                "due to permission error. Check the permissions of the directory and "
+                "try again. If the issue persists, consider cleaning up the directory manually."
```

Comment:
let's include the error in the message.

Reply:
We raise the exception in this case, so no need to include it in the message?